src/java.base/share/classes/java/security/PEMRecord.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.security;
+
+import jdk.internal.javac.PreviewFeature;
+
+import sun.security.util.Pem;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * {@code PEMRecord} is a {@link DEREncodable} that represents Privacy-Enhanced
+ * Mail (PEM) data by its type and Base64 form.  {@link PEMDecoder} and
+ * {@link PEMEncoder} use {@code PEMRecord} when representing the data as a
+ * Java API cryptographic object is not desired or there is no other
+ * {@code DEREncodable} for the type.
+ *
+ * <p>{@code PEMRecord} may have a null {@code type} and {@code pem} when
+ * there is no PEM data.  If there is PEM data, {@code type} and
+ * {@code pem} will both be non-null. {@code leadingData} may be null if the
+ * decoded data only contains PEM data. All values can never be null.
+ *
+ * <p> During the instantiation of this record, there is no validation for the
+ * {@code type} or {@code pem}.
```

Comment:
The ctors throw `IllegalArgumentException` though. I think you need to be more specific about what type of validation you mean here.